### PR TITLE
Add new modals based on #406

### PR DESCRIPTION
**Pull request by howard-e**
This adds 3 new modals based on https://github.com/w3c/aria-at-app/issues/406.

* 'Assistive Technology and Browser Details Modal' as `AtAndBrowserDetailsModal`, along with logic to account for described scenarios.
* 'Warning Modal' and 'Danger Modal' as `BasicThemedModal`, where a theme ('warning' or 'danger'), content and possible actions may be passed as props.
* 'New AT Version Modal' and 'Edit an Existing AT Version Modal' as `UpdateVersionModal`.

See usage examples below:
```javascript
<AtAndBrowserDetailsModal
    show={isShowing}
    isAdmin={false}
    atName="NVDA" // passed from Test Queue
    atVersion="2022.3.5" // passed from Test Queue
    atVersions={['2022.3.5', '2022.3.7', '2023.23.5']}
    browserName="Chrome" // passed from Test Queue
    browserVersion="97.0.4758"
    browserVersions={[
        '96.0.4758',
        '97.0.4758',
        '98.0.4758',
        '100.0.4896'
    ]}
    patternName="Checkbox" // required if isAdmin=true
    testerName="alflennik" // required if isAdmin=true
    handleAction={handleAction} // expects func(updatedAtVersion, updatedBrowserVersion) but to be discussed
    handleClose={handleClose}
/>

<BasicThemedModal
    show={isShowing}
    theme="warning" // or danger
    title="Your AT version is different than the one used to record this result"
    dialogClassName="modal-50w"
    content={
        <>
            You are currently running <b>NVDA 2022.2</b>, but
            are editing a test result that was submitted with{' '}
            <b>NVDA 2021.4</b>.
            <br />
            <br />
            Do you want to update the AT version used to record
            this test result?
        </>
    } // possible content to be templated and passed when required
    actionButtons={[
        {
            text: 'Update AT Version',
            action: handleUpdateAtVersion
        },
        {
            text: 'Continue without changes',
            action: handleContinue
        }
    ]}
    handleClose={handleClose}
/>

<UpdateVersionModal
    show={isShowing}
    title="Edit NVDA Version 2022"
    actionType="edit" // default is add
    handleAction={handleAction} // expects func(actionType, { updatedVersionText, updatedDateAvailabilityText }) but to be discussed
    handleClose={handleClose}
/>
```

### CODE REVIEW

File: client/components/common/UpdateVersionModal/index.jsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                            value={versionText}
+                            onChange={handleVersionTextChange}
+                        />
+                    </Form.Group>
+
+                    <Form.Group>
+                        <Form.Label>
+                            Approximate date of availability
+                        </Form.Label>
+                        <Form.Control
```
Comment (howard-e) [inline client/components/common/UpdateVersionModal/index.jsx:65]: This input should be masked to a date format (`DD-MM-YYYY`?)

File: client/components/common/AtAndBrowserDetailsModal/index.jsx [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+                                    </option>
+                                ))}
+                            </Form.Control>
+                        </Form.Group>
+
+                        {/* Tester Scenario 7 */}
+                        {uaMajor === '0' && (
+                            <Form.Group className="at-browser-details-full-column">
+                                <Form.Label>Enter Browser Version</Form.Label>
+                                <Form.Control
```
Comment (howard-e) [inline client/components/common/AtAndBrowserDetailsModal/index.jsx:391]: This version string input could be masked (`x.y.z-additional.identifiers`?)

File: client/components/common/UpdateVersionModal/index.jsx (hunk 1 above)
Comment (evmiguel) [reply client/components/common/UpdateVersionModal/index.jsx:65]: +1 to this

**Review by evmiguel — APPROVED** [review]
Great job on all three modals and on updating the `BasicModal`. The usage examples demonstrate clean APIs for these components. The `AtAndBrowserDetailsModal`, especially, is clean, easy to read, and well implemented. Looking forward to seeing these components in the wild!

**Review by alflennik — APPROVED** [review]

**Pull request merged**